R/LSBuffonsneedlemanipulation.R

```diff
@@ -0,0 +1,166 @@
+                            lowerCI = CI95lower, Median = med,   upperCI = CI95upper))
+  jaspResults[["summaryTable"]] <- summaryTable
+  
+  ## 1. prior and posterior plot for proportion of crosses
+  if (plot2){
+    propPlot <- createJaspPlot(title = paste("Prior and Posterior for Proportion of Crosses"),
+                               width = 480, height = 320)
+    propPlot$position <- 2
+    propPlot$dependOn(c("k", "n", "a", "b", "length", "CI", "plot2"))
+    propPlot$addCitation("JASP Team (2018). JASP (Version 0.9.2) [Computer software].")
+    
+    # values
+  
+    xValue <- seq(0,1,0.005)
+    propPost <- dbeta(xValue, a + crosses, b + observations - crosses)
+    propPrior <-dbeta(xValue, a, b)
+    dataProp = data.frame(values = c(xValue, xValue),
+                          density = c(propPost, propPrior),
+                          group = c(rep("Posterior",201), rep("Prior",201))
+    )
+    # axis specification
+    propPlot0 <- ggplot2::ggplot(data = dataProp,  ggplot2::aes(x = values, y = density)) +
+      ggplot2::xlab("Proportion of Crosses") +
+      ggplot2::ylab("Density") #+
+    #ggplot2::coord_cartesian(xlim = c(0, 1), ylim = c(0, 1.6*max(propPost)))
+    
+    # fill in the plot
+    propPlot$plotObject <- jaspGraphs::themeJasp(propPlot0) +
+      ggplot2::geom_line(ggplot2::aes(linetype = group), size = 1) +
+      ggplot2::scale_linetype_manual("", values = c("Posterior" = "solid",
+                                                    "Prior" = "dashed")) 
+    if (legend1){
+      propPlot$plotObject <-  propPlot$plotObject + 
+        ggplot2::theme(legend.position = c(.17, .9))
+    }
+    jaspResults[["propPlot"]] <- propPlot
+  }
+  
+  
+  ## 2. Distribution Plot
+  if (plot3){
+    distPlot <- createJaspPlot(title = paste("Implied Prior and Posterior for", "\u03c0"),  width = 480, height = 320)
+    distPlot$position <- 3
+    distPlot$dependOn(c("k", "n", "a", "b", "length", "CI", "plot3", "CIArrow"))
+    distPlot$addCitation("JASP Team (2018). JASP (Version 0.9.2) [Computer software].")
+    
+    # values
+    x <- seq(2,4,0.01)
+    yPost <- 2 * l / (x^2 * d) * dbeta((2 * l / (x * d)), a + crosses, b + observations - crosses)
+    yPrior <- 2 * l / (x^2 * d) * dbeta((2 * l / (x * d)), a, b)
+    yPi <- seq(0, 1.6*max(yPost), 1.6*max(yPost)/99)
+    
+    data = data.frame(values = c(x, x, rep(pi, 100)),
+                      density = c(yPost, yPrior, yPi),
+                      group = c(rep("Implied Posterior",201), rep("Implied Prior",201), rep("\u03c0", 100))
+    )
+    data$group<-factor(data$group, levels=c("Implied Posterior","Implied Prior","\u03c0"))
+    
+    # axis specification
+    distPlot0 <- ggplot2::ggplot(data = data,  ggplot2::aes(x = values, y = density)) +
+      ggplot2::ggtitle("") + # for , pi
+      ggplot2::xlab("\u03c0") +
+      ggplot2::ylab("Density") +
+      ggplot2::coord_cartesian(xlim = c(2, 4), ylim = c(0, 1.6*max(yPost)))
```

Comment:
Can we achieve the same thing with plot editing?

Reply:
Well, plot editing could do that, but it's better to have sensible default axis limits IMO. If people would need to always use plot editing for a particular plot, the defaults of the plots are just off.

Reply:
Yeah I tried 10 times again with the default setting and all fall in the axis limits. Only extreme cases require plot editing. Shall I just keep the limit unchanged?

Reply:
I get that in your particular example the posterior is *usually* in that range, but I would suggest to reconsider whether it's realistic to expect that the users will specify the output in the intended way, or whether they will try to change input just to see how the analysis behaves. But if you really think it's not a biggie and 99% of the time the user interacts with your analysis and it will not make a difference whether or not the axis is fixed, you can of course decide to leave it at that.

Reply:
I also discussed with EJ long before and he said it looks simpler in the GUI. Also most of the time the distribution falls in the default range so that it is not quite necessary to add additional inputs. I would just leave it like that for now.

Reply:
Right. Just to point out, this is not an issue of the GUI, I am not asking you to add input fields that specify the range of the axis, but to use some heuristic that programmatically determines the plotting range based on the location and shape of the distribution you want to display.